Issue/13401 jetpack my site section

malinajirka opened the pull request:
Partially fixes #13401

This PR 
- introduces "Jetpack" section/label on the My Site fragment
- adds a new "Jetpack settings" row to the menu on Jetpack sites
- fixes toolbar style on "Jetpack Settings" screen when accessed from gutenberg
- adds "jetpack_settings_viewed" tracking event
- removes "white listed ip addresses" from the JetpackSecuritySettingsFragment, since they were never implemented. The fragment contains just a button which doesn't do anything on click.

To test:

###### Self-hosted site (without Jetpack)
1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible

###### Jetpack Site as admin 
1. Verify "Jetpack" label is visible and "Jetpack settings" row is also visible
2. Click on "Jetpack settings" and notice the jetpack settings screen is shown
3. Verify "jetpack_settings_viewed" has been tracked
4. Go back to My Site
5. Click on Site Settings
6. Scroll down to Security settings
7. Open security settings and verify "jetpack_settings_viewed" has been tracked

###### Jetpack Site as Author
1. Verify "Jetpack" label is visible and "Jetpack settings" row is **not** visible
Note: There is a bug and Stats never load in this scenario. Reported in https://github.com/wordpress-mobile/WordPress-Android/issues/13476

###### Jetpack Site, but using self-hosted login flow
1. Verify "Jetpack" label is visible and "Jetpack settings" row is **not** visible

###### Atomic Site
1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible
 
###### .com site with business plan (not atomic site)
1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible

###### .com site with non-business plan
1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible

<img width="904" alt="Screenshot 2020-11-25 at 12 56 56" src="https://user-images.githubusercontent.com/2261188/100224892-b9a3aa00-2f1d-11eb-90b5-8bcb31ecaf63.png">


----------------------------------------

###### Gutenberg unsupported block (more info https://github.com/wordpress-mobile/WordPress-Android/pull/13011)
1. Create a post with an unsupported block on the web - on a Jetpack site
2. Go to the app -> Jetpack settings -> disable "Allow WordPress.com login"
3. Go to Post List -> open the post from step 1
4. Click on the question mark on the unsupported block
5. Click on "Open Jetpack settings"
6. Notice the settings are shown, "jetpack_settings_viewed" has been tracked and everything looks ok

<img width="370" alt="Screenshot 2020-11-17 at 12 36 41" src="https://user-images.githubusercontent.com/2261188/99385650-93ed2400-28d1-11eb-8792-02bfb4662510.png">


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

malinajirka committed:
Introduce Jetpack section on My Site screen

malinajirka committed:
Start Jetpack security settings from ActivityLauncher

malinajirka committed:
Show Jetpack section on MySite screen only on jetpack sites

malinajirka committed:
Fix toolbar style

malinajirka committed:
Move Plans above Jetpack section on My Site screen

malinajirka committed:
Remove logic which changed Jetpack section visibility on My Site

malinajirka committed:
Hide jetpack settings on non-jetpack sites

malinajirka commented on WordPress/src/main/res/layout/my_site_fragment.xml:
```diff
@@ -372,6 +372,39 @@
 
                     </com.google.android.material.card.MaterialCardView>
 
+                    <!--Plan-->
```
- moved "Plan" section to the top 
- introduced "Jetpack" label
- introduced "Jetpack Settings" row

malinajirka commented on WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java:
```diff
@@ -964,6 +965,18 @@ public static void viewNotificationsSettings(Activity activity) {
         activity.startActivity(intent);
     }
 
+    public static void viewJetpackSecuritySettings(Activity activity, SiteModel site) {
```
Minor refactoring since the EditPostActivity was not using ActivityLauncher.

malinajirka committed:
Hide jetpack label on self-hosted sites

malinajirka committed:
Merge branch 'develop' into issue/13401-jetpack-my-site-section-2

malinajirka committed:
Show jetpack label on My Site screen only for pure jetpack sites

malinajirka committed:
Remove whitelisted ip addresses option from jetpack security settings

malinajirka committed:
Add jetpack_settings_viewed tracking event

ashiagr commented:
Nice work @malinajirka ! Code-wise all looks good to me. 

These test cases passed for me:

> ###### Jetpack Site (self-hosted with Jetpack installed and connected)
> 1. Verify "Jetpack" label is visible and "Jetpack settings" row is also visible ✅
> 2. Click on "Jetpack settings" and notice the jetpack settings screen is shown ✅
> 3. Verify "jetpack_settings_viewed" has been tracked ✅

> ###### .com site with business plan (not atomic site)
> 1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible ✅

> ###### .com site with non-business plan
> 1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible ✅

> Gutenberg unsupported block ✅

-----------

### Findings/ Questions

> ###### Self-hosted site (without Jetpack)
> 1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible 

**Question** Should "Jetpack settings" be visible when we login to Jetpack connected self-hosted site using site address? It looks like a logged-out experience and I can see "Install Jetpack" in the Notifications tab + "Jetpack settings" on My Site. Isn't "Jetpack settings" confusing with the "Install Jetpack" option?

> ###### Atomic Site
> 1. Verify "Jetpack" label is **not** visible and "Jetpack settings" row is **not** visible 

**Question** This site is shown as atomic in SA, yet I see "Jetpack" label and "Jetpack settings" row. Is that expected? Or could it be a site-configuration issue at my side?

<img width="391" alt="Screenshot 2020-11-25 at 1 54 10 PM" src="https://user-images.githubusercontent.com/1405144/100201173-bc0a0200-2f25-11eb-8a8f-99a32edb19a2.png">
 
![atomic_site](https://user-images.githubusercontent.com/1405144/100201677-55391880-2f26-11eb-8077-7b56b10545f9.png)

malinajirka committed:
Hide jetpack section on atomic sites

malinajirka committed:
Hide jetpack section on jetpack site when using self-hosted login

malinajirka committed:
Hide jetpack settings for users without permissions

malinajirka committed:
Show jetpack label when user doesn't have settings permissions

malinajirka committed:
Show jetpack label on self-hosted site when jetpack connected

malinajirka committed:
Fix jetpack label on atomic sites

malinajirka committed:
Fix lint

malinajirka commented:
Thanks so much for the review! Nice catch ⭐ 

I've fixed the issues, refactored the code and updated the test scenarios. I'm sorry I didn't have it right in the first run. Let me know if you find anything else. Thanks!

ashiagr commented:
> I've fixed the issues, refactored the code and updated the test scenarios. I'm sorry I didn't have it right in the first run. 

Please don't be sorry, 😉 and thanks for detailed testing instructions, they're super helpful!

Everything works great for me now 👍 except that for "Jetpack Site as Author" scenario, I can see "Jetpack" label and "Jetpack settings" row.  If it helps, I can add you as author to this site. Lmk.

malinajirka commented:
> except that for "Jetpack Site as Author" scenario, I can see "Jetpack" label and "Jetpack settings" row. If it helps, I can add you as author to this site. Lmk.

The "Jetpack" label should be visible in this scenario (there was a mistake in the testing instructions 😞 ). However, "Jetpack settings" row shouldn't be visible. I just re-tested it with two accounts and I don't see the row. I'll give it a try on your site.

ashiagr reviewed (approved):
Thanks for giving it a try 👍. It was a specific case with that email account. Tested with another account and it worked as expected. Approving it based on our discussion on Slack.
